java/CMakeLists.txt

```diff
@@ -27,18 +31,29 @@ if(JNI_FOUND)
     ${src_base}/vowpalWabbit_learner_VWScalarLearner.cc
     ${src_base}/vowpalWabbit_learner_VWScalarsLearner.cc
     ${src_base}/vowpalWabbit_VW.cc
+    ${src_base}/jni_spark_vw.cc
+    ${src_base}/vector_io_buf.cc
+    ${src_base}/jni_spark_cluster.cc
+    ${src_base}/util.cc
+    ${CMAKE_CURRENT_SOURCE_DIR}/../vowpalwabbit/spanning_tree.cc
   )
 
   add_library(vw_jni SHARED ${vw_jni_headers} ${vw_jni_sources})
   target_link_libraries(vw_jni PUBLIC vw)
-  target_include_directories(vw_jni PUBLIC ${CMAKE_CURRENT_SOURCE_DIR} ${JNI_INCLUDE_DIRS})
+  target_include_directories(vw_jni PUBLIC 
+    ${CMAKE_CURRENT_SOURCE_DIR} 
+    ${JNI_INCLUDE_DIRS} 
+    ${CMAKE_CURRENT_SOURCE_DIR}/../vowpalwabbit)
```

Comment:
The fact that vw_jni links against vw should mean that the headers come in with the target. Does it still work if you remove this?

Reply:
Ping on this.  (I'm looking to make a release soon and would like to merge...)

Reply:
I can spent some time tomorrow on it.

Reply:
removed, still works. thanks!